third_party/txt/src/minikin/FontCollection.cpp

```diff
@@ -528,8 +528,27 @@ void FontCollection::itemize(const uint16_t* string,
           }
           start -= prevChLength;
         }
-        result->push_back(
-            {family->getClosestMatch(style), static_cast<int>(start), 0});
+        result->push_back({family->getClosestMatch(style, ch, variant),
+                           static_cast<int>(start), 0});
+        run = &result->back();
+        lastFamily = family.get();
+      } else if (family.get() == lastFamily &&
+                 // We need to change this condition to more general logic. And
+                 // when changing the font, it is necessary to check whether the
+                 // existing characters also have glyphs.
+                 !(U_GET_GC_MASK(prevCh) & U_GC_L_MASK) &&
+                 (U_GET_GC_MASK(ch) & U_GC_L_MASK)) {
+        size_t start = utf16Pos;
+        auto current_font = family->getClosestMatch(style, ch, variant);
+        if (result->back().fakedFont.font != current_font.font) {
+          const size_t prevChLength = U16_LENGTH(prevCh);
+          run->end -= prevChLength;
+          if (run->start == run->end) {
+            result->pop_back();
+          }
+          start -= prevChLength;
+        }
```

Comment:
This is the logic taken from the logic that handles the emoji above. 
What we are doing here is to check ,ultimately,  if we can draw a previous glyph using the font that draws the current glyph if certain conditions are met.

Reply:
From what I can see in the comment for the original code that you copied from, the if block is supposed to be a workaround for a combined mark+emoji modifier character. (Here is its use case: https://emojipedia.org/emoji-modifier-sequence, just in case.) Are you sure the code acts as you expected? Wouldn't the rest of the code without the if block just do?

> if we can draw a previous glyph using the font that draws the current glyph

If so, how about a previous-previous glyph? Are you taking account of various cases?

Reply:
@swift-kim 
I think... this was added because of my misunderstanding about [this](https://github.com/flutter-tizen/engine/pull/106#issuecomment-859200687).
I guessed there is a problem that the character styles of sentences specified in one style seem to be different when too many fonts are unintentionally selected even though they belong to the same font family. But this seems to be all wrong.

Basically, if some fonts belongs to the same font family, there should be no awkwardness when displayed. I think this problem is also due to the font for the emoji too, So now I think that this is an unnecessary if statement

@bwikbs I've already checked about the above, but could you please check it again and update the PR?

Reply:
@swift-kim  Hmm... I'll update this part after I talk to @bbrto21 to figure out my understanding is correct.
Anyway, thanks for the thoughtful review!

Reply:
> Are you sure the code acts as you expected? 

Yes, it works exactly as intended. However, I agree that this solution itself is a workaround for our specific situation and is no longer needed(As @bbrto21  pointed out...).

> If so, how about a previous-previous glyph? Are you taking account of various cases?

You pointed out a weakness of this workaround! :smile:  At first I wanted to check the whole string, but it seemed too complicated and would cause performance issue. And I thought that our current situation is a font config setting problem, I am going to end it at this level. (Yes, this is an excuse for why this workaround came out.)

Anyway, I will remove this block and move the condition to the above.

Reply:
@swift-kim @bbrto21 I've update code! How about now?

Reply:
Looks good to me!